src/main/scala/org/constellation/p2p/Cluster.scala

```diff
@@ -235,6 +237,11 @@ class Cluster[F[_]: Concurrent: Timer: ContextShift](ipManager: IPManager[F], da
           val authSignRequest = PeerAuthSignRequest(Random.nextLong())
           val req = client.postNonBlockingF[F, SingleHashSignature]("sign", authSignRequest)(C)
 
+          checkBalanceJoiningPeer(client).map(isValid => {
```

Comment:
Perhaps ifM here. Also, since that method is getting pretty long, why don't we move the req.flatTap into another method and call that within ifM